manage-operational-layers/src/main/java/com/esri/arcgismaps/sample/manageoperationallayers/components/MapViewModel.kt

```diff
@@ -0,0 +1,173 @@
+
+    // a list of the active map image layer names
+    var layerNames = mutableStateListOf<String>()
+        private set
+
+    // a list of the inactive map image layer names
+    var inactiveLayers = mutableStateListOf<Layer>()
+        private set
+
+    init {
+        // set the three map image layers
+        val imageLayerElevation = ArcGISMapImageLayer(
+            url = application.getString(R.string.elevationServiceURL)
+        )
+        val imageLayerCensus = ArcGISMapImageLayer(
+            url = application.getString(R.string.censusServiceURL)
+        )
+        val imageLayerDamage = ArcGISMapImageLayer(
+            url = application.getString(R.string.damageServiceURL)
+        )
+        // get a list of the layer names
+        layerNames.addAll(
+            listOf(
+                imageLayerElevation.name,
+                imageLayerCensus.name,
+                imageLayerDamage.name
+            )
+        )
+
+        // add the layers to the map's operational layers
+        mapViewState.arcGISMap.apply {
+            operationalLayers.addAll(
+                listOf(
+                    imageLayerElevation,
+                    imageLayerCensus,
+                    imageLayerDamage
+                )
+            )
+        }
+    }
+
+    /**
+     * Swap the active layer with the layer on top.
+     */
+    fun moveLayerUp(layerName: String) {
+        // get a copy of the operational layers
+        val operationalLayers = mapViewState.arcGISMap.operationalLayers.toMutableList()
+        // if move up on the first item is selected, then return
+        if (operationalLayers.first().name == layerName) {
+            return
+        }
+        // get the index of the tapped layer
+        val layerIndex = operationalLayers.indexOf(operationalLayers.find { it.name == layerName })
+        // swap the selected layer with the layer on top
+        operationalLayers.swap(layerIndex, layerIndex - 1)
+        // update the layer names list
+        layerNames.apply {
+            clear()
+            addAll(operationalLayers.map { layer -> layer.name })
+        }
+        // update the operational layers
+        mapViewState.arcGISMap.operationalLayers.apply {
+            clear()
+            addAll(operationalLayers)
+        }
+    }
+
+    /**
+     * Swap the active layer with the layer on bottom.
+     */
+    fun moveLayerDown(layerName: String) {
+        // get a copy of the operational layers
+        val operationalLayers = mapViewState.arcGISMap.operationalLayers.toMutableList()
+        // if move down on the last item is selected, then return
+        if (operationalLayers.last().name == layerName) {
+            return
+        }
+        // get the index of the tapped layer
+        val layerIndex = operationalLayers.indexOf(operationalLayers.find { it.name == layerName })
+        // swap the selected layer with the layer on bottom
+        operationalLayers.swap(layerIndex, layerIndex + 1)
```

Comment:
Related to my other comment, is this not moving the layer _up_? Since it is moving to a higher index it will now display above the layer to its right in the list.

Reply:
Moving the layer "down" in the UI corresponds to moving the layer down the operational layers which visually will display the layer on top of the other layers in the list. 

I agree that ideally, the first layer on the operational layers should be the top layer displayed but that is not the behavior of the API. As the sample is "Manage operational layers" my opinion was to follow the order of layers defined in the API.